libs/blockchain/src/blockchain.service.ts

```diff
@@ -0,0 +1,218 @@
+/* eslint-disable new-cap */
+/* eslint-disable no-underscore-dangle */
+import fs from 'fs';
+import { Inject, Injectable, OnApplicationBootstrap } from '@nestjs/common';
+import { Keyring } from '@polkadot/api';
+import { Call } from '@polkadot/types/interfaces';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+import { IMethod } from '@polkadot/types/types';
+import { Vec } from '@polkadot/types';
+import { FrameSystemEventRecord } from '@polkadot/types/lookup';
+import { HexString } from '@polkadot/util/types';
+import blockchainConfig, { addressFromSeedPhrase, IBlockchainConfig } from './blockchain.config';
+import Redis from 'ioredis';
+import { InjectRedis } from '@songkeys/nestjs-redis';
+import { BlockchainRpcQueryService } from './blockchain-rpc-query.service';
+
+export const NONCE_SERVICE_REDIS_NAMESPACE = 'NonceService';
+
+export type Sr25519Signature = { Sr25519: HexString };
+interface SIWFTxnValues {
+  msaId: string;
+  address: string;
+  handle: string;
+  newProvider: string;
+}
+
+export interface ICapacityInfo {
+  providerId: string;
+  currentBlockNumber: number;
+  nextEpochStart: number;
+  remainingCapacity: bigint;
+  totalCapacityIssued: bigint;
+  currentEpoch: number;
+}
+
+/**
+ * To be able to provide mostly unique nonces to submit transactions on chain we would need to check a number of
+ * temporarily locked keys on redis side and get the first available one. This number defines the number of keys
+ * we should look into before giving up
+ */
+const NUMBER_OF_NONCE_KEYS_TO_CHECK = 50;
+
+/**
+ * Nonce keys have to get expired shortly so that if any of nonce numbers get skipped we would still have a way to
+ * submit them after expiration
+ */
+const NONCE_KEY_EXPIRE_SECONDS = 2;
+const CHAIN_NONCE_KEY = 'chain:nonce';
+
+function getNonceKey(suffix: string) {
+  return `${CHAIN_NONCE_KEY}:${suffix}`;
+}
```

Comment:
These are moved into `redis-keys.constants.ts` please use those instead of duplicating here